apps/meteor/client/views/room/Sidepanel/SidepanelItem/SidePanelTagIcon.tsx

```diff
@@ -0,0 +1,13 @@
+import { Box, Icon } from '@rocket.chat/fuselage';
+import type { ComponentProps, ReactElement } from 'react';
+import { isValidElement } from 'react';
+
+const SidePanelTagIcon = ({ icon }: { icon: ReactElement | Pick<ComponentProps<typeof Icon>, 'name' | 'color'> | null }) => {
+	if (isValidElement(icon)) {
+		return <Box>{icon}</Box>;
+	}
+
+	return icon && typeof icon === 'object' && 'name' in icon ? <Icon size='x12' mie={4} {...icon} /> : null;
```

Comment:
here too, can't we be more specific about the types instead of doing all these checks?